server/src/main/java/org/elasticsearch/search/fetch/subphase/highlight/HighlightField.java

```diff
@@ -101,14 +76,14 @@ public void writeTo(StreamOutput out) throws IOException {
     public static HighlightField fromXContent(XContentParser parser) throws IOException {
         ensureExpectedToken(XContentParser.Token.FIELD_NAME, parser.currentToken(), parser);
         String fieldName = parser.currentName();
-        Text[] fragments = null;
+        Text[] fragments;
         XContentParser.Token token = parser.nextToken();
         if (token == XContentParser.Token.START_ARRAY) {
             List<Text> values = new ArrayList<>();
             while (parser.nextToken() != XContentParser.Token.END_ARRAY) {
                 values.add(new Text(parser.text()));
             }
-            fragments = values.toArray(new Text[values.size()]);
+            fragments = values.toArray(Text.EMPTY_ARRAY);
```

Comment:
I think t's a good default to use the constant when we have it anyway. Not that it matters too much here, but it's generally slightly faster and saves memory in case `values` is empty.